src/ajax/xhr.js

```diff
@@ -88,12 +88,15 @@ jQuery.ajaxTransport( function( options ) {
 									xhrSuccessStatus[ xhr.status ] || xhr.status,
 									xhr.statusText,
 
-									// Support: IE9
-									// Accessing binary-data responseText throws an exception
-									// (#11426)
-									typeof xhr.responseText === "string" ? {
-										text: xhr.responseText
-									} : undefined,
+									// XHR2 responseText throws an exception on binary data,
+									// return it raw and let a converter or caller handle it
+									// https://xhr.spec.whatwg.org/#the-responsetext-attribute
+									// (trac-11426, gh-2498)
+									xhr.responseType === "arraybuffer" ||
```

Comment:
That will force the `json` and `document` (XML) cases down the wrong branch, or at least wrong for the current logic in ajax.js that wants to convert the `text` result explicitly rather than letting XHR do it. This was due to various bugs and/or inability to control cases like [trac-13276](http://bugs.jquery.com/ticket/13276). @jaubourg says in that ticket he could only make it happen in IE9 so maybe we could hew closer to the spec once we drop IE9 support after 3.0? At that point we should be able to drop the `typeof` check as well and just look at `.responseType`.